orc/src/main/java/com/netflix/iceberg/orc/TypeConversion.java

```diff
@@ -0,0 +1,172 @@
+      case INTEGER:
+        return TypeDescription.createInt();
+      case LONG:
+        return TypeDescription.createLong();
+      case FLOAT:
+        return TypeDescription.createFloat();
+      case DOUBLE:
+        return TypeDescription.createDouble();
+      case DATE:
+        return TypeDescription.createDate();
+      case TIME:
+        return TypeDescription.createInt();
+      case TIMESTAMP:
+        return TypeDescription.createTimestamp();
+      case STRING:
+        return TypeDescription.createString();
+      case UUID:
+        return TypeDescription.createBinary();
+      case FIXED:
+        return TypeDescription.createBinary();
+      case BINARY:
+        return TypeDescription.createBinary();
+      case DECIMAL: {
+        Types.DecimalType decimal = (Types.DecimalType) type;
+        return TypeDescription.createDecimal()
+            .withScale(decimal.scale())
+            .withPrecision(decimal.precision());
+      }
+      case STRUCT: {
+        TypeDescription struct = TypeDescription.createStruct();
+        for(Types.NestedField field: type.asStructType().fields()) {
+          columnIds.add(field.fieldId());
+          struct.addField(field.name(), toOrc(field.type(), columnIds));
+        }
+        return struct;
+      }
+      case LIST: {
+        Types.ListType list = (Types.ListType) type;
+        columnIds.add(list.elementId());
+        return TypeDescription.createList(toOrc(type.asListType().elementType(),
+            columnIds));
+      }
+      case MAP: {
+        Types.MapType map = (Types.MapType) type;
+        columnIds.add(map.keyId());
+        TypeDescription key = toOrc(map.keyType(), columnIds);
+        columnIds.add(map.valueId());
+        return TypeDescription.createMap(key,
+            toOrc(type.asMapType().valueType(), columnIds));
+      }
+      default:
+        throw new IllegalArgumentException("Unhandled type " + type.typeId());
+    }
+  }
+
+  /**
+   * Convert an ORC schema to an Iceberg schema.
+   * @param schema the ORC schema
+   * @param columnIds the column ids
+   * @return the Iceberg schema
+   */
+  public Schema fromOrc(TypeDescription schema, int[] columnIds) {
+    return new Schema(convertOrcToType(schema, columnIds).asStructType().fields());
+  }
+
+  Type convertOrcToType(TypeDescription schema, int[] columnIds) {
+    switch (schema.getCategory()) {
+      case BOOLEAN:
+        return Types.BooleanType.get();
+      case BYTE:
+      case SHORT:
+      case INT:
+        return Types.IntegerType.get();
+      case LONG:
+        return Types.LongType.get();
+      case FLOAT:
+        return Types.FloatType.get();
+      case DOUBLE:
+        return Types.DoubleType.get();
+      case STRING:
+      case CHAR:
+      case VARCHAR:
+        return Types.StringType.get();
+      case BINARY:
+        return Types.BinaryType.get();
+      case DATE:
+        return Types.DateType.get();
+      case TIMESTAMP:
```

Comment:
Does ORC differentiate between with zone and without zone? I think we need support for both. We can work around it by keeping Iceberg metadata, but I'd rather have everything represented correctly in data files.

Reply:
Not yet. I have jira open to add it.

Reply:
Great, what's the JIRA?

Reply:
https://issues.apache.org/jira/browse/ORC-189